h/streamer/views.py

```diff
@@ -21,6 +21,19 @@ def websocket_view(request):
     if origin is not None:
         if origin != request.host_url and origin not in allowed:
             return httpexceptions.HTTPForbidden()
+
+    # Provide environment which the WebSocket handler can use...
+    request.environ.update({
+        'h.ws.authenticated_userid': request.authenticated_userid,
+        'h.ws.effective_principals': request.effective_principals,
+        'h.ws.registry': request.registry,
+        'h.ws.streamer_work_queue': streamer.WORK_QUEUE,
+    })
```

Comment:
No, I like this in general, I just don't like having to pass the entire application registry in. Not sure there's a lot we can do about that right now, though.